Mark provided Google dependencies as `compileOnly`

Stephan202 opened the pull request:
This way at runtime `reactor-error-prone` uses the classes provided by Error Prone. This makes the project also compatible with newer versions of Error Prone.

Stephan202 committed:
Mark provided Google dependencies as `compileOnly`

This way at runtime `reactor-error-prone` uses the classes provided by
Error Prone. This makes the project also compatible with newer versions
of Error Prone.

Stephan202 commented:
Force-pushed a change to fix the `reactor-error-prone` tests; I missed that those were broken by the original change. There's now still a build failure in the `reactor-sample` module, but my lack of Gradle expertise prevents me from solving it:
```
> Task :reactor-sample:compileJava FAILED
error: plug-in not found: ErrorProne
```
@lhotari do you know how to resolve this?

(I tried some of the suggestions found online, but none seems to apply (properly?) to this project's build.)

lhotari commented:
Thanks for the PR @Stephan202 . I'll take a look at the issue in tests.

lhotari commented:
@Stephan202  the `plug-in not found: ErrorProne` issue would get resolved by adding 
```
errorprone 'com.google.errorprone:error_prone_core:2.3.3'
```
to `dependencies` in `reactor-sample/build.gradle`

Stephan202 commented:
Rebased and force-pushed that one change in there. Confirmed that this fixes it, tnx! (I thought I tried this, but perhaps I added it to the wrong module :thinking:. Oh well.)

lhotari commented:
0.1.4 is now available with the fix.
